docs/user-guide/src/bmo/introduction.md

```diff
@@ -1,23 +1,77 @@
-3. user supplied: Fields such as `image` are supplied by user to dictate boot image for the server.
-4. dynamic fields: Fields such as IP could be dynamically assigned to the server at run time by DHCP server.
+## Enrolling BareMetalHosts
 
-During the life cycle of a bare metal host, upgrade is one example, some of these fields change with information coming from Ironic or other controllers while fields, such as MAC address, do not change (upgrade is one example).
+To enroll a bare-metal machine as a `BareMetalHost`, you need to know at least
+the following properties:
 
-BMO can also work with the Cluster API Provider Metal3 (CAPM3) controller. With the involvement of CAPM3 and Ironic, a simplified information flow path and an overview of the BareMetalHost resource is shown below:
+1. The IP address and credentials of the BMC - the remote management controller
+   of the host.
+2. The protocol that the BMC understands. Most common are IPMI and Redfish.
+3. Boot technology that can be used with the host and the chosen protocol.
+   Most hardware can use network booting, but some Redfish implementations also
+   support virtual media (CD) boot.
+4. MAC address that is used for booting. **Important:** it's a MAC address of
+   an actual NIC of the host, not the BMC MAC address.
+5. The desired boot mode: UEFI or legacy BIOS. UEFI is the default and should
+   be used unless there are serious reasons not to.
+
+This is a minimal example of a valid BareMetalHost:
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: node-0
+  namespace: metal3
+spec:
+  bmc:
+    address: ipmi://192.168.111.1:6230
+    credentialsName: node-0-bmc-secret
+  bootMACAddress: 00:5a:91:3f:9a:bd
+  online: true
+```
+
+When this resource is created, it will undergo *inspection* that will populate
+more fields as part of the `status`.
+
+## Deploying BareMetalHosts
+
+To provision a bare-metal machine, you will need a few more properties:
+
+1. The URL and checksum of the image. Images should be in QCOW2 or raw format.
+   It is common to use various cloud images with BMO, e.g.
+   [Ubuntu](https://cloud-images.ubuntu.com/) or
+   [CentOS](https://cloud.centos.org/centos/). **Important:** not all images
+   are compatible with UEFI boot - check their description.
+2. Optionally, user data: a secret with a configuration or a script that is
+   enterpreted by the first-boot service embedded in your image. The most
+   common service is
+   [cloud-init](https://cloudinit.readthedocs.io/en/latest/index.html), some
+   distributions use [ignition](https://coreos.github.io/ignition/).
+3. Optionally, network data: a secret with the network configuration that is
+   enterpreted by the first-boot service. In some cases, the network data is
+   embedded in the user data instead.
+
+Here is a complete example of a host that will be provisioned with a CentOS 8
```

Comment:
while at it you can also update this and the yaml below to CentOS 9 perhaps, just a nit